docs/composer-studio/developer-guides/cx-framework/integration/remix.md

```diff
@@ -0,0 +1,105 @@
+
+### `AtamaClient#action(args, pathOverride)`
+
+Run an action based on the current path and form data passed in.
+
+| Parameter | Type | Default | Required? | Description |
+|:----------|:-----|:--------|:----------|:------------|
+| args | `DataFunctionArgs` | - | Yes | A remix loader function object. [Remix loader documentation](https://remix.run/docs/en/main/route/loader), [Interface definition](https://github.com/remix-run/remix/blob/9a9c955f241c5742d227542e13c7387b99d4628b/packages/remix-server-runtime/routeModules.ts#L17-L21) |
+| pathOverride | `string` | - | No | An optional path to load instead of whatever is coming in through `loaderArgs.request` |
+
+The FormData must include at least the following key/value pairs:
+
+| Key | Type | Default | Required? | Description |
+|:----------|:-----|:--------|:----------|:------------|
+| actionKey | `string` | - | Yes | The key of the action on the component type. This is required to understand which action to run. |
+| componentType | `string` | - | Yes | The component type this action is supposed to run on. |
+
+Returns a [JSON response](https://remix.run/docs/en/1.14.0/utils/json).
+
+## Action Configuration
+
+The action configuration is an object to list the resolver methods for each component type, separated by "read" and "write" actions.
+
+Example configuration
+
+```js
+const actionConfig = {
+  read: {
+    cart: {
+      "get-cart": {
+        input: () => {},
+        output: () => {},
+      }
+    }
+  },
+  write: {
+    "product-detail": {
+      "add-to-cart": {
+        input: () => {},
+        output: () => {},
+      }
+    },
+  }
+}
+```
+
+| Parameter | Type | Default | Required? | Description |
+|:----------|:-----|:--------|:----------|:------------|
+| read | `Record<string, Record<string, ActionDefinition>>` | - | No | An object of component types and their respective action resolvers. |
+| write | `Record<string, Record<string, ActionDefinition>>` | - | No | An object of component types and their respective action resolvers. |
+
+The `ActionDefinition` is defined like this:
+
+| Parameter | Type | Default | Required? | Description |
+|:----------|:-----|:--------|:----------|:------------|
+| input | `(data: any, dataFnArgs: DataFunctionArgs) => Promise<object>` | - | Yes | A method that accepts the action input data and potentially transforms it. It's either a pass-through or a modification of the input data. |
+| write | `(data: any, request: Request, inputResult: any) => Promise<object | [object, ResponseInit]>` | - | Yes | A method that accepts the Action Business Capability response and either passes it through or modifies it based on the request and the result of the `input` method. |
+
+## Cache
+
+Remix leaves caching up to user-land. Because we're fetching experiences that we can consider cachable we provide a simple hook to implement whatever caching is desirable for the specific solution.
```

Comment:
What's "user-land"?

Reply:
Sorry that's a weird way of saying Remix "doesn't do caching" but instead a developer using Remix has to implement their own caching - e.g. by setting headers that are interpreted by a PaaS, or a client, **or** they have to implement some other caching mechanism (potentially even on top of headers).